robot/impl/local_robot.go

```diff
@@ -1113,7 +1129,7 @@ func (r *localRobot) applyLocalModuleVersions(cfg *config.Config) {
 	}
 }
 
-func (r *localRobot) reconfigure(ctx context.Context, newConfig *config.Config, forceSync bool) {
+func (r *localRobot) reconfigureInLock(ctx context.Context, newConfig *config.Config, forceSync bool) {
```

Comment:
I can't comment off-diff. I'm not sure if there's a thing to do in this PR, or whether it's worth postponing. 

But now that we have a lock for reconfigure, do we still need the `localRobot.reconfiguring` variable that this function sets to true while it's running? If all of the readers of that variable are now taking the `reconfigurationLock`, we can presumably remove it altogether.

Same question for the `localRobot.configRevisionMu`. And maybe `localRobot.lastWeakDependentsRound`?

As I said, perfectly happy to postpone investigation in this PR. I very much want this change to go in. But I would like to be in agreement about how suspicious we are that those untouched variables still require special synchronization primitives.

Reply:
`localRobot.reconfiguring` is still used as part of determining whether it's alright for viam agent to restart viam server, so that one, at least, is probably still necessary. You raise good points about the other two, though; I'd be happy to look at those as part of a separate ticket/PR.

Reply:
Do we still need to call this method `reconfigureInLock`, though? That name now confuses me with the presence of `reconfigurationLock`.

Reply:
renamed back to `reconfigure`, I think we need the variables because the readers of the variables do not always take the lock. having the configRevisionMu means we won't have to wait for reconfiguration to complete to get the config revision. lastWeakDependentsRound is still needed to prevent extra weak dependent updates